geostreams/gltg-bmp
# Clowder Extractor & Docker Setup

**Main things that have changed:**
- Added an extractor using SimpleExtractor named bmp.practices_extractor.
   - Created a copy of the import script and changed it a bit to get states and assumptions data from api instead of file.
- Fixed a bug in import script where it failed if missing_hucs log file didnt exist
- Refactored docker-compose files to enable mounting files to extractor and also for ease
- Added dockerfile and docker-compose entry for flask api as well. The files should be updated automatically in container without having to recreate the container.
- Updated readme to add instructions and details
- Setup traefik routes for clowder and api.

**How to test:**
- Run `docker-compose up` to see if all the containers are running up properly and check the routes.
- Test the extractor by uploading file to clowder. I tested it with a subset of the practices file and since I made minimal changes to the import_script, the test file should be in similar format (Same sheet name and header row) as the original practices file. We can change this if we need to adjust for other formats as well. 


**Issue Link:**
[GLGVO-701](https://opensource.ncsa.illinois.edu/jira/browse/GLGVO-701)
[GLGVO-706](https://opensource.ncsa.illinois.edu/jira/browse/GLGVO-706)
